cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -280,83 +287,160 @@ static String removeReplicasFromConfig(String throttleConfig, Set<String> replic
-    ConfigEntry currLeaderThrottle = brokerConfigs.get(LEADER_REPLICATION_THROTTLED_RATE_CONFIG);
-    ConfigEntry currFollowerThrottle = brokerConfigs.get(FOLLOWER_REPLICATION_THROTTLED_RATE_CONFIG);
-    List<AlterConfigOp> ops = new ArrayList<>();
-    if (currLeaderThrottle != null) {
-      if (currLeaderThrottle.source().equals(ConfigEntry.ConfigSource.STATIC_BROKER_CONFIG)) {
-        LOG.debug("Skipping removal for static leader throttle rate: {}", currFollowerThrottle);
-      } else {
-        LOG.debug("Removing leader throttle rate: {} on broker {}", currLeaderThrottle, brokerId);
-        ops.add(new AlterConfigOp(new ConfigEntry(LEADER_REPLICATION_THROTTLED_RATE_CONFIG, null), AlterConfigOp.OpType.DELETE));
-      }
+  private void applyIncrementalAlterConfigsForTopics(Map<ConfigResource, Collection<AlterConfigOp>> bulkOps)
+    throws ExecutionException, InterruptedException, TimeoutException {
+    if (bulkOps == null || bulkOps.isEmpty()) {
+      return;
     }
-    if (currFollowerThrottle != null) {
-      if (currFollowerThrottle.source().equals(ConfigEntry.ConfigSource.STATIC_BROKER_CONFIG)) {
-        LOG.debug("Skipping removal for static follower throttle rate: {}", currFollowerThrottle);
-      } else {
-        LOG.debug("Removing follower throttle rate: {} on broker {}", currFollowerThrottle, brokerId);
-        ops.add(new AlterConfigOp(new ConfigEntry(FOLLOWER_REPLICATION_THROTTLED_RATE_CONFIG, null), AlterConfigOp.OpType.DELETE));
+    AlterConfigsResult result = _adminClient.incrementalAlterConfigs(bulkOps);
+    Map<ConfigResource, KafkaFuture<Void>> futures = result.values();
+    Map<ConfigResource, Collection<AlterConfigOp>> succeeded = new HashMap<>();
+    for (Map.Entry<ConfigResource, KafkaFuture<Void>> entry : futures.entrySet()) {
+      ConfigResource cf = entry.getKey();
+      KafkaFuture<Void> future = entry.getValue();
+      try {
+        future.get(CLIENT_REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        succeeded.put(cf, bulkOps.get(cf));
+      } catch (Exception e) {
+        if (!topicExists(cf.name())) {
+          LOG.debug("Skipping configs for non-existent topic {} (confirmed via listTopics)", cf.name());
+          continue;
+        }
+        throw e;
       }
     }
-    if (!ops.isEmpty()) {
-      changeBrokerConfigs(brokerId, ops);
+    if (!succeeded.isEmpty()) {
+      waitForConfigs(succeeded);
     }
   }
 
+  private Map<ConfigResource, Config> getBrokerConfigs(Set<Integer> brokerIds)
+    throws ExecutionException, InterruptedException, TimeoutException {
+    if (brokerIds == null || brokerIds.isEmpty()) {
+      return Collections.emptyMap();
+    }
+    List<ConfigResource> resources = brokerIds.stream()
+      .map(id -> new ConfigResource(ConfigResource.Type.BROKER, String.valueOf(id)))
+      .collect(Collectors.toList());
+    return _adminClient.describeConfigs(resources).all().get(CLIENT_REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS);
```

Comment:
Would it make sense to batch the requests here? Maybe it would make more sense to just call this method with a subset of the brokerIds in the cluster for when there is a large number of brokers in the cluster?